src/reference/asciidoc/kafka.adoc

```diff
@@ -1073,6 +1074,18 @@ See its javadocs for more information.
 A retry adapter is not provided for any of the batch <<message-listeners, message listeners>> because the framework has no knowledge of where, in a batch, the failure occurred.
 Users wishing retry capabilities, when using a batch listener, are advised to use a `RetryTemplate` within the listener itself.
 
+[[stateful-retry]]
+===== Stateful Retry
+
+It is important to understand that the retry discussed above suspends the consumer thread (if a `BackOffPolicy` is used); there is no interaction with the broker during retry.
+For lengthy retry sequences, if the `max.poll.interval.ms` property is exceeded, the broker will think the consumer is dead and will revoke the assigned partitions and perform a rebalance.
```

Comment:
I'd suggest rephrasing it in order to differentiate the `max.poll.interval.ms` timeout (the server says "you've spent too much time consuming, I'm giving the partition to someone else") and the `session.timeout.ms` (the server didn't receive any heartbeat from the consumer and considers it as a dead one).

Example:
```
For lengthy retry sequences, if the `max.poll.interval.ms` property is exceeded (which means that consumer is processing the record for too long), the broker will revoke the assigned partitions and perform a rebalance. This can cause another available consumer to consume the record, thus producing duplicates.
```

Reply:
Thanks; yes, this assumes users already understand the difference between `max.poll.interval.ms` and `session.timeout.ms`.